fast_barcode_scanner/android/src/main/kotlin/com/jhoogstraat/fast_barcode_scanner/cache/ImageHelper.kt

```diff
@@ -0,0 +1,102 @@
+import android.content.Context
+import android.graphics.ImageFormat
+import android.graphics.Rect
+import android.graphics.YuvImage
+import android.media.Image
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+
+class ImageHelper {
+    private var savedCodes: MutableMap<String, String> = mutableMapOf()
+
+    private object Holder {
+        val INSTANCE = ImageHelper()
+    }
+
+    companion object {
+        @JvmStatic
+        fun getInstance(): ImageHelper {
+            return Holder.INSTANCE
+        }
+    }
+
+    // Store image to the path with barcode as filename
+    private fun storeImage(imageBytes: ByteArray?, key: String, context: Context) {
+        val externalFilesDirectory = context.cacheDir;
+        val imageFile: File
+
+        try {
+            externalFilesDirectory.mkdirs()
+            val barcodeDirectory = File(externalFilesDirectory, "barcode_images")
+            if (!barcodeDirectory.exists()) {
+                barcodeDirectory.mkdirs()
+            }
+
+            imageFile = File.createTempFile(key, ".jpeg", barcodeDirectory)
+            imageFile.deleteOnExit()
+        } catch (e: IOException) {
+            throw RuntimeException(e)
+        }
+
+        try {
+            FileOutputStream(imageFile).use { fos ->
+                fos.write(imageBytes)
+            }
+            savedCodes[key] = imageFile.absolutePath
+        } catch (e: IOException) {
+            e.printStackTrace()
+        }
+    }
+
+    private fun isImageSaved(code: String): Boolean {
+        return savedCodes.contains(code)
+    }
+
+    // Retrieve image from cache by barcode
+    fun retrieveImagePath(code: String): String? {
+        return savedCodes[code]
+    }
+
+    fun clearCache(context: Context) {
+        savedCodes.clear()
+        val externalFilesDirectory = context.cacheDir
+        val barcodeDirectory = File(externalFilesDirectory, "barcode_images")
+        if (barcodeDirectory.exists()) {
+            barcodeDirectory.deleteRecursively()
+        }
+    }
+
+    fun storeImageToCache(image: Image, code: String, context: Context) {
```

Comment:
we probably need to do this operation in a background thread. https://developer.android.com/kotlin/coroutines.

Making the saving of files async might introduce a possibility for a race condition (Flutter might ask for the image before it has been saved successfully). We should be sure we can handle this kind of situation.